ERCS/erc-7546.md

```diff
@@ -0,0 +1,178 @@
+- **Function-Level Upgradeability**: One Proxy contract corresponds to multiple Implementation contracts, basically each responsible for a specific function.
+- **Factory**: A contract that clones Proxies with a common Implementation(s). In the context of upgradeability, it allows for the simultaneous upgrade of these cloned Proxies.
+
+Here are the use cases:
+
+1. For basic needs without Upgradeability or a Factory, *Regular smart contract deployment* suffices.
+2. When a Factory is needed without Upgradeability, [ERC-1167](./eip-1167.md) is suitable.
+3. For Contract-Level Upgradeability without a Factory, [ERC-1822](./eip-1822.md) can be used.
+4. For Contract-Level Upgradeability with a Factory, the Beacon from [ERC-1967](./eip-1967.md) is applicable.
+5. For Function-Level Upgradeability without a Factory, [ERC-2535](./eip-2535.md) is available.
+6. For Function-Level Upgradeability with a Factory, this ***Upgradeable Clone Standard*** is the ideal choice.
+
+![Fig. Use Cases](../assets/eip-7546/images/usecases.png)
+
+
+## Specification
+> The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+In the EVM, contract accounts are characterized by four primary fields: *nonce*, *balance*, *code*, and *storage*. This ERC's architecture modularizes these functionalities into three distinct types of contracts, each serving a specific purpose when combined to represent a single account:
+
+1. **Proxy Contract**: Maintains the state of the contract account, such as nonce, balance, and storage. This contract delegatecalls to the `Function Contract` as registered in the `Dictionary Contract`, ensuring the state and logic are separated but effectively integrated.
+2. **Dictionary Contract**: Acts as a dispatcher that routes function calls based on their selectors to the appropriate `Function Contract`. It manages the dynamic aspects of contract behavior, facilitating function upgrades and dynamic addressing. By externalizing this contract from the `Proxy Contract`, it becomes factory/clone-friendly and supports simultaneous upgradeability.
+3. **Function (Implementation) Contract**: Implements the executable logic for function calls. When delegatecalled by the `Proxy Contract`, it performs the actual computations or logic as defined in the contract's code.
+
+This architecture not only aligns with the core attributes of an EVM contract account but also significantly enhances the modularity, upgradeability, and scalability of smart contracts by clarifying account state, function dispatching, and logic implementation.
+
+### Proxy Contract
+This contract requests the `Dictionary Contract` to retrieve the associated `Function Contract` address based on its function selector, and then delegatecall to it.
```

Comment:
Please only use backticks (\`) for inline code snippets. When referencing a specifically defined term (eg. Dictionary Contract) use \_italics\_ or \*\*bold\*\*.

Reply:
Understood. I will make the adjustments for readable text.